contracts/utils/ERC721BatchTransfer.sol

```diff
@@ -0,0 +1,159 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+/**
+ * @title ERC721 Batch Transfer
+ *
+ * @notice ETransfer ERC721 tokens in batches to a single wallet or multiple wallets. This supports ERC721M and ERC721CM contracts.
+ * @notice To use any of the methods in this contract the user has to approve this contract to control their tokens using either `setApproveForAll` or `approve` functions from the ERC721 contract.
+ */
+contract ERC721BatchTransfer {
+    error InvalidArguments();
+    error NotOwnerOfToken();
+    error InvalidCaller();
+
+    event BatchTransferToSingle(
+        address indexed contractAddress,
+        address indexed to,
+        uint256 amount
+    );
+
+    event BatchTransferToMultiple(
+        address indexed contractAddress,
+        uint256 amount
+    );
+
+    constructor() {}
+
+    modifier noZero() {
+        if (msg.sender == address(0)) revert InvalidCaller();
```

Comment:
what is the purpose of this? how could msg.sender ever be the null address?

Reply:
It's not null address. Address(0) is valid address - we just don't know who has / will have the private key. Just a practice to not allow address(0) as it's known to be used as "null".

Reply:
it is a valid address but nobody has the compute power to generate the private key for address(0) which is why it used as the burn address

it currently holds > $200M in assets
https://etherscan.io/address/0x0000000000000000000000000000000000000000

if we are using this internally then it doesnt matter, just calling out that its not a necessary check to have

Reply:
Ok just remove the check to save more gas.